### DISCUSSION TITLE
Streaming with short lived connections is not working

### DISCUSSION THREAD

**Discussion by antxv** [deepgram/community#390] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

Hi,

I own a voice dictation application that supports multiple recognition services. I am attempting to add deepgram as one of the recognition options that my users can select. I am getting data from the user's microphone and sending it to the service. Because this is a dictation scenario, each utterance from the user may be short, and there may be long periods of time in between utterances. I also need to control endpointing manually. My application is written in Go, so I am using your Go package. 

When the user begins speaking, my application does the following:
1. open the streaming connection
2. start sending the utterance to the service in 100ms chunks. 

When the user finishes speaking, my application does the following:
1. stop sending chunks.
2. request the transcript.
3. close the connection. 
 
For short (~2 seconds) utterances, requesting the transcript hangs for several seconds and eventually returns the error:

`websocket: close 1011 (internal server error): Deepgram did not receive audio data or a text message within the timeout window. See https://dpgr.am/net0001. `

I am certain that I am sending data before requesting the transcript, because of my logging and successful behavior with other recognizers.

Here are my transcription options:

```
client.LiveTranscription(
			deepgram.LiveTranscriptionOptions{
				Model:            "general",
				Tier:             "nova",
				Language:         "en-US",
				Punctuate:        false,
				Profanity_filter: false,
				Smart_format:     false,
				FillerWords:      "true",
				Alternatives:     5,
				Numerals:         false,
				Interim_results:  false,
				Endpointing:      "false",
				Encoding:         "linear16",
				Sample_rate:      16000,
			},
		)
```

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

wss://api.deepgram.com/v1/listen?alternatives=5&encoding=linear16&endpointing=false&filler_words=true&language=en-US&model=general&sample_rate=16000&tier=nova

**Comment by adeschamps** [comment, marked as answer]
With an utterance that short, the Deepgram server would be waiting to receive more audio before it has enough context to transcribe (how much is enough depends on multiple factors, such as duration thresholds, and endpointing, and more). If you want to tell Deepgram that you won't be sending any more audio, send a WebSocket text frame containing `{"type": "CloseStream"}`. This will cause Deepgram to transcribe any audio that's been received (even though we might not have received the "optimal" amount of context yet), send back the transcript, and _then_ close the WebSocket. See [the docs](https://developers.deepgram.com/reference/streaming#close-stream) for more (although that's basically all there is to it!)

**Reply by antxv** [reply]
Thank you, that fixed it. Note that the documentation uses single quotes, which fail to be parsed. The double quotes in your example above work fine.
